### PR TITLE
Add coverage CI job containing full CKB VM test suite

### PR DESCRIPTION
**Pull request by xxuejie**
This is adapted from #66 thanks to @u2's help!

I originally rejected the idea of adding coverage job in #66 but upon further consideration, that's a mistake, we should add coverage job in this repo since:

* Cross-posting coverage report from another repository is very tricky to get right
* While test suite repo contains a lot of useful and complete tests, the tests included in this repo is a good supplement for invalid cases
* With CI job here, bugs in PRs can be detected earlier

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -50,7 +50,71 @@ matrix:
...
+            - libgmp-dev
+            - gawk
+            - libtool
+            - patchutils
+            - libexpat-dev
+            - zlib1g-dev
+      script:
+        git clone --recursive https://github.com/nervosnetwork/ckb-vm-test-suite &&
+        ln -snf .. ckb-vm-test-suite/ckb-vm &&
+        sudo docker run --rm -v `pwd`:/code xxuejie/riscv-gnu-toolchain-rv64imac:xenial-20190606 cp -r /riscv /code/riscv &&
```
Comment (u2) [inline .travis.yml]: Why use `sudo` here?

Comment (xxuejie) [reply .travis.yml]: I believe docker requires sudo on Linux?

**Comment by doitian** [comment]
Not if the user has been added to group docker

**Comment by u2** [comment]
> # [Codecov](https://codecov.io/gh/nervosnetwork/ckb-vm/pull/70?src=pr&el=h1) Report
> > Merging [#70](https://codecov.io/gh/nervosnetwork/ckb-vm/pull/70?src=pr&el=desc) into [develop](https://codecov.io/gh/nervosnetwork/ckb-vm/commit/9e128d4502f112a883f4447d29c0c3e5f5e87bf1?src=pr&el=desc) will **increase** coverage by `10.13%`.
> > The diff coverage is `n/a`.
> 
> [Impacted file tree graph](https://codecov.io/gh/nervosnetwork/ckb-vm/pull/70?src=pr&el=tree)
> 
> ```diff
> @@             Coverage Diff              @@
> ##           develop      #70       +/-   ##
> ============================================
> + Coverage    82.92%   93.06%   +10.13%     
> ============================================
>   Files            3       24       +21     
>   Lines           41     3403     +3362     
> ============================================
> + Hits            34     3167     +3133     
> - Misses           7      236      +229
> ```
> 
> [Impacted Files](https://codecov.io/gh/nervosnetwork/ckb-vm/pull/70?src=pr&el=tree)	Coverage Δ	
> [binary/src/interpreter32.rs](https://codecov.io/gh/nervosnetwork/ckb-vm/pull/70/diff?src=pr&el=tree#diff-YmluYXJ5L3NyYy9pbnRlcnByZXRlcjMyLnJz)		
> [binary/src/asm64.rs](https://codecov.io/gh/nervosnetwork/ckb-vm/pull/70/diff?src=pr&el=tree#diff-YmluYXJ5L3NyYy9hc202NC5ycw==)		
> [binary/src/interpreter64.rs](https://codecov.io/gh/nervosnetwork/ckb-vm/pull/70/diff?src=pr&el=tree#diff-YmluYXJ5L3NyYy9pbnRlcnByZXRlcjY0LnJz)		
> [src/memory/mod.rs](https://codecov.io/gh/nervosnetwork/ckb-vm/pull/70/diff?src=pr&el=tree#diff-c3JjL21lbW9yeS9tb2QucnM=)	`96.66% <0%> (ø)`	
> [src/instructions/common.rs](https://codecov.io/gh/nervosnetwork/ckb-vm/pull/70/diff?src=pr&el=tree#diff-c3JjL2luc3RydWN0aW9ucy9jb21tb24ucnM=)	`100% <0%> (ø)`	
> [src/instructions/rvc.rs](https://codecov.io/gh/nervosnetwork/ckb-vm/pull/70/diff?src=pr&el=tree#diff-c3JjL2luc3RydWN0aW9ucy9ydmMucnM=)	`89.18% <0%> (ø)`	
> [src/machine/mod.rs](https://codecov.io/gh/nervosnetwork/ckb-vm/pull/70/diff?src=pr&el=tree#diff-c3JjL21hY2hpbmUvbW9kLnJz)	`96.29% <0%> (ø)`	
> [src/memory/sparse.rs](https://codecov.io/gh/nervosnetwork/ckb-vm/pull/70/diff?src=pr&el=tree#diff-c3JjL21lbW9yeS9zcGFyc2UucnM=)	`92.56% <0%> (ø)`	
> [tests/test_asm.rs](https://codecov.io/gh/nervosnetwork/ckb-vm/pull/70/diff?src=pr&el=tree#diff-dGVzdHMvdGVzdF9hc20ucnM=)	`98.72% <0%> (ø)`	
> [src/instructions/register.rs](https://codecov.io/gh/nervosnetwork/ckb-vm/pull/70/diff?src=pr&el=tree#diff-c3JjL2luc3RydWN0aW9ucy9yZWdpc3Rlci5ycw==)	`97.28% <0%> (ø)`	
> ... and [17 more](https://codecov.io/gh/nervosnetwork/ckb-vm/pull/70/diff?src=pr&el=tree-more)		
> [Continue to review full report at Codecov](https://codecov.io/gh/nervosnetwork/ckb-vm/pull/70?src=pr&el=continue).
> 
> > **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> > `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> > Powered by [Codecov](https://codecov.io/gh/nervosnetwork/ckb-vm/pull/70?src=pr&el=footer). Last update [9e128d4...141c986](https://codecov.io/gh/nervosnetwork/ckb-vm/pull/70?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

Should exclude test files. 

```
--exclude-path=path     comma-separated paths to exclude from the coverage
                         report
```

**Comment by xxuejie** [comment]
> Should exclude test files.
> 
> ```
> --exclude-path=path     comma-separated paths to exclude from the coverage
>                          report
> ```

Already excluded.

**Pull request merged**